domains/client/domain-executor/src/aux_schema.rs

```diff
@@ -128,6 +140,39 @@ where
     load_decode(backend, execution_receipt_key(block_hash).as_slice())
 }
 
+pub(super) fn track_domain_hash_to_primary_hash<Backend, Hash, PHash>(
```

Comment:
why track them when you can already get them from the runtime?

Reply:
I don't know it's possible to get the corresponding primary block hash given the domain block hash from the runtime. Are you sure about that?

Reply:
I was mostly referring to this function 
https://github.com/subspace/subspace/blob/main/crates/pallet-receipts/src/lib.rs#L237
if we modify this and accept an execution receipt hash, we can fetch that, but I see that would need some work.
Ignore my suggestion if this is unnecessary 👍🏼

Reply:
Relying on the receipt from the runtime requires the corresponding receipt is able to be submitted successfully in time (it may fail due to some bug as we have already seen :D) otherwise the domain block processing will be blocked, we probably don't want that.